M2/Macaulay2/m2/matrix2.m2

```diff
@@ -24,7 +24,49 @@ smithNormalForm = method(
      	  KeepZeroes => true
 	  }
      )
-smithNormalForm Matrix := o -> (f) -> (
+smithNormalForm Matrix := o -> A -> (
+    (D',P,Q) := preSmithNormalForm(A,o);
+    D := mutableMatrix D';
+    -- get into a diagonal matrix
+    n := min(numRows D, numColumns D);
+    start'i := 0;
+    done := false;
+    P' := mutableMatrix map((ring D')^(numRows D'));
+    Q' := mutableMatrix map((ring D')^(numColumns D'));
+    while not done do (
+	pos'i := position(start'i..(n-2),j->D_(j+1,j+1)!=0 and D_(j+1,j+1) % D_(j,j) != 0);	
+	if pos'i === null then (
+	    if start'i > 0 then start'i = 0 else done = true
+	    ) else (
+	    i := pos'i + start'i;
+	    start'i = i + 1;
+	    a := D_(i,i);
+	    b := D_(i+1,i+1);
+	    (g,e,f) := toSequence gcdCoefficients(a,b);
```

Comment:
As of recently (#3597), `gcdCoefficients` now returns a sequence, so calling `toSequence` is redundant.